pandas/core/generic.py

```diff
@@ -4506,7 +4506,15 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
 
     def copy(self, deep=True):
         """
-        Make a copy of this objects data.
+        Make a copy of this objects indices and data.
+
+        Deep copy (default) will create a new object with a copy of the objects
```

Comment:
Given these start to dive into the parameters and implementations I think they are better served in a dedicated Notes section